### PR TITLE
expression: modify the test framework for vectorized expression evaluation functions

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve? 
Modify the test framework for vectorized expression evaluation functions.

### What is changed and how it works?
Create a test file for each different kind of vectorized expression. Add corresponding cases to each individual test file. Pass the corresponding cases to the unit test and benchmark functions in bench_test.go in each test file. It can effectively reduce conflicts.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/bench_test.go [hunk 1]
```diff
@@ -385,9 +369,9 @@ func genVecExprBenchCase(ctx sessionctx.Context, funcName string, testCase vecEx
 	return expr, input, output
 }
 
-func (s *testEvaluatorSuite) TestVectorizedEvalOneVec(c *C) {
+func (s *testEvaluatorSuite) testVectorizedEvalOneVec(c *C, vecExprCases vecExprBenchCases) {
```
Comment (SunRunAway) [inline expression/bench_test.go]: How about adding some comments for contributors with those four functions/methods.

File: expression/bench_test.go [hunk 2]
```diff
@@ -385,9 +369,9 @@ func genVecExprBenchCase(ctx sessionctx.Context, funcName string, testCase vecEx
...
 	ctx := mock.NewContext()
-	for funcName, testCases := range vecExprBenchCases {
+	for funcName, testCases := range vecExprCases {
```
Comment (qw4990) [inline expression/bench_test.go]: How about supporting specifying test case according to a system environment variable?
For example:
```
>: set vec_test_func_sig=builtinAddDateAndDurationSig
>: go test -v
```
Then only the `builtinAddDateAndDurationSig` will be processed.

Comment (SunRunAway) [reply expression/bench_test.go]: By running `go test -run=^$ -bench=BenchmarkVectorizedBuiltinMathEvalOneVec`, is it ok after this pr?

Comment (Reminiscent) [reply expression/bench_test.go]: > By running `go test -run=^$ -bench=BenchmarkVectorizedBuiltinMathEvalOneVec`, is it ok after this pr?

It's ok.

File: expression/bench_test.go (hunk 1 above)
Comment (Reminiscent) [reply expression/bench_test.go]: update. PTAL

**Pull request merged**